File: Python/Sorting/quicksort.py
def quickSort(arr):
    length = len(arr)
    if length < 2:
        return arr

    lastValue = arr[-1]
    leftArray = []
    rightArray = []
    for i in range(0, length-1):
        if(arr[i] < lastValue):
             leftArray.append(arr[i])
        else:
            rightArray.append(arr[i])

    return quickSort(leftArray)+[lastValue]+quickSort(rightArray)


arr = [5, 10, 1, 9, 2, 8, 3, 7, 4, 6, 1];
print(quickSort(arr));

File: Javascript/quicksort.js

quickSort = arr => {
  let length = arr.length;
  if(length < 2) {
    return arr;
  }

  let lastIndex = length - 1;
  let lastValue = arr[lastIndex];

  let leftArray = [];
  let rightArray = [];

  for (var i = 0; i < lastIndex; i++) {
    if(lastValue > arr[i]) {
      leftArray.push(arr[i]);
    } else {
      rightArray.push(arr[i]);
    }
  }

  return [...quickSort(leftArray), lastValue, ...quickSort(rightArray)];

}
const numbers = [8, 5, 6, 9, 3, 1, 4, 2, 7, 10]
console.log(quickSort(numbers));

// describe("first",() =>{
//   it("second",() => {
//     expect([ 1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ]).toEqual(quickSort(numbers));
//   })
// })

// jasmine quicksort.js
